main.tf

```diff
@@ -106,133 +231,342 @@ data "aws_iam_policy_document" "secrets_access" {
-  health_check_grace_period     = var.health_check_grace_period
-  tags                          = local.tags
-}
-
-module "autoscaling" {
-  source             = "github.com/byu-oit/terraform-aws-app-autoscaling?ref=v1.0.1"
-  app_name           = var.app_name
+    actions = ["sts:AssumeRole"]
+  }
+}
+resource "aws_iam_role" "codedeploy_role" {
+  name                 = "${var.app_name}-codedeploy-role"
+  permissions_boundary = var.role_permissions_boundary_arn
+  assume_role_policy   = data.aws_iam_policy_document.codedeploy_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "AWSCodeDeployRole" {
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
+  role       = aws_iam_role.codedeploy_role.name
+}
+
+resource "aws_codedeploy_deployment_group" "deploymentgroup" {
+  app_name               = aws_codedeploy_app.app.name
+  deployment_group_name  = "${var.app_name}-deployment-group"
+  service_role_arn       = aws_iam_role.codedeploy_role.arn
+  deployment_config_name = "CodeDeployDefault.ECSAllAtOnce"
+
+  ecs_service {
+    cluster_name = aws_ecs_cluster.cluster.name
+    service_name = aws_ecs_service.service.name
+  }
+
+  blue_green_deployment_config {
+    deployment_ready_option {
+      action_on_timeout = "CONTINUE_DEPLOYMENT"
+    }
+    terminate_blue_instances_on_deployment_success {
+      action                           = "TERMINATE"
+      termination_wait_time_in_minutes = 15
+    }
+  }
+
+  deployment_style {
+    deployment_option = "WITH_TRAFFIC_CONTROL"
+    deployment_type   = "BLUE_GREEN"
+  }
+  auto_rollback_configuration {
+    enabled = true
+    events  = ["DEPLOYMENT_FAILURE"]
+  }
+
+  load_balancer_info {
+    target_group_pair_info {
+      prod_traffic_route {
+        listener_arns = [aws_alb_listener.https.arn]
+      }
+      //      TODO test listener traffic
+      //      test_traffic_route {
+      //        listener_arns = aws_alb_listener.....
+      //      }
+      target_group {
+        name = aws_alb_target_group.blue.name
+      }
+      target_group {
+        name = aws_alb_target_group.green.name
+      }
+    }
+  }
+
+  lifecycle { ignore_changes = [blue_green_deployment_config] }
+}
+
+# ==================== CloudWatch ====================
+resource "aws_cloudwatch_log_group" "container_log_group" {
+  name              = local.cloudwatch_log_group_name
+  retention_in_days = var.log_retention_in_days
+  tags              = var.tags
+}
+
+# ==================== AutoScaling ====================
+resource "aws_appautoscaling_target" "main" {
   min_capacity       = var.min_capacity
   max_capacity       = var.max_capacity
-  resource_id        = "service/${module.fargate.ecs_cluster.name}/${module.fargate.ecs_service.name}"
+  resource_id        = "service/${aws_ecs_cluster.cluster.name}/${aws_ecs_service.service.name}"
   scalable_dimension = "ecs:service:DesiredCount"
   service_namespace  = "ecs"
-  step_up = {
+}
+resource "aws_appautoscaling_policy" "up" {
```

Comment:
I'm not sure how I feel about about the autoscaling policy. I'd love to get Josh's feedback on having this specific autoscaling policy be the only option.

Reply:
I'm more worried about the metric_alarms than the policies.

Reply:
Fair. I guess I am just wondering about the whole approach.

Reply:
I'm just putting something down now. I didn't want to customize this yet due to how crazy it can be. This is why I asked if we wanted to bake in autoscaling or allow the user to do it themselves via the provider resources specifically.

Reply:
Done